src/lib/generateDefinition.ts

```diff
@@ -4,30 +4,53 @@ import {getClusterAttributeValue} from './utils';
 import * as m from './modernExtend';
 import * as zh from 'zigbee-herdsman/dist';
 import {philipsLight} from './philips';
+import {Device} from 'zigbee-herdsman/dist/controller/model';
 
-interface GeneratedExtend {extend: ModernExtend, source: string, lib?: string}
+interface GeneratedExtend {extend?: ModernExtend, extendFn?: (a: object) => ModernExtend, args?: object, source: string, lib?: string}
 type ExtendGenerator = (endpoint: Zh.Endpoint) => Promise<GeneratedExtend[]>;
 type Extender = [string[], ExtendGenerator];
 
-function generateSource(device: Zh.Device, generatedExtend: GeneratedExtend[]): string {
+function generateSource(zigbeeModel: string, definition: Definition, generatedExtend: GeneratedExtend[]): string {
```

Comment:
`zigbeeModel` here would not be necessary, but TS fails to understand that `definition` contains `zigbeeModel` field, thus failing the build. 

If there is a better way to fix it - would be glad to do it.

Reply:
That's because `definition` does not always have a `zigbeeModel`, I suggest to create a new local type here:

```
type DefinitionWithZigbeeModel = Definition & {zigbeeModel: string[]};
```

Reply:
Thank you. Updated.